### PR TITLE
Support SHOW CATALOGS syntax in Spark extensions

### PR DESCRIPTION
**Pull request by pan3793**
### _Why are the changes needed?_

Spark support multi-catalogs since 3.0, and it's important to Kyuubi as Kyuubi aim to be a Gateway of LakeHouse.

The implementation is different from SPARK-35973 (available since 3.3.0).

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by turboFei** [comment]
Looks that we can also add a `show sessions` syntax to show the  openSessions in current spark engine?

File: extensions/spark/kyuubi-extension-spark-common/src/main/scala/org/apache/spark/sql/catalog/ShowCatalogsCommand.scala [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+case class ShowCatalogsCommand(pattern: Option[String]) extends RunnableCommand {
+  override val output: Seq[Attribute] = Seq(
+    AttributeReference("catalog", StringType, nullable = false)())
+
+  // The implementation use eager strategy to list catalog, which is different from SPARK-35973
+  override def run(sparkSession: SparkSession): Seq[Row] = {
+    val configuredCatalogs = sparkSession.sessionState.conf.getAllConfs.keys
+      .filter { _ startsWith "spark.sql.catalog." }
+      .map { _ stripPrefix "spark.sql.catalog." }
+      .filterNot { _ contains "." }
```
Comment (ulysses-you) [inline extensions/spark/kyuubi-extension-spark-common/src/main/scala/org/apache/spark/sql/catalog/ShowCatalogsCommand.scala:40]: what if the catalog has multi-identifier, a.b.c

Comment (pan3793) [reply extensions/spark/kyuubi-extension-spark-common/src/main/scala/org/apache/spark/sql/catalog/ShowCatalogsCommand.scala:40]: good question, after reading spark code, I think `a.b.c` is a legal catalog name. let me try

**Comment by pan3793** [comment]
We have a previous discussion https://github.com/apache/incubator-kyuubi/issues/929, eagerly load catalogs is more intuitive

File: extensions/spark/kyuubi-extension-spark-common/src/main/scala/org/apache/spark/sql/catalog/ShowCatalogsCommand.scala (hunk 1 above)
Comment (pan3793) [reply extensions/spark/kyuubi-extension-spark-common/src/main/scala/org/apache/spark/sql/catalog/ShowCatalogsCommand.scala:40]: it's legal,   as long as we use back quote "use \`p.g\`"

Comment (turboFei) [reply extensions/spark/kyuubi-extension-spark-common/src/main/scala/org/apache/spark/sql/catalog/ShowCatalogsCommand.scala:40]: but it seems that `spark.sql.catalog.`p.g`` also contains "."

Comment (pan3793) [reply extensions/spark/kyuubi-extension-spark-common/src/main/scala/org/apache/spark/sql/catalog/ShowCatalogsCommand.scala:40]: yea, i see. i'm thinking if we can do a light check and cache the result to implement the eager load.

**Pull request closed without merging**